gui/src/components/SettingsWallet.tsx

```diff
@@ -1,28 +1,39 @@
 import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  Button,
+  FormControl,
+  FormControlLabel,
+  FormHelperText,
+  FormLabel,
+  Radio,
+  RadioGroup,
+  Stack,
+  TextField,
+} from "@mui/material";
 import { invoke } from "@tauri-apps/api/tauri";
+import { map } from "lodash";
+import React from "react";
 import { useCallback, useEffect, useState } from "react";
-import { FieldValues, useForm } from "react-hook-form";
+import { Controller, FieldValues, useForm } from "react-hook-form";
 
-import { useInvoke } from "../../hooks/tauri";
-import { Address, Wallet, walletSchema } from "../../types";
-import Button from "../Base/Button";
-import { FieldRadio, FieldText } from "./Fields";
+import { useInvoke } from "../hooks/tauri";
+import { Address, Wallet, walletSchema } from "../types";
 
-export function WalletSettings() {
+export function SettingsWallet() {
   const { data: wallet, mutate } = useInvoke<Wallet>("get_wallet");
 
   const {
     register,
     handleSubmit,
     reset,
-    formState: { isValid, dirtyFields, errors },
+    formState: { isValid, isDirty, dirtyFields, errors },
```

Comment:
@gabrielpoca I'm not sure if you saw the github issue linked a few lines below?

https://github.com/react-hook-form/react-hook-form/issues/3213

I was having problems with `isDirty` not working properly with nested objects. that's why `isDirtyAlt` is there instead.

Reply:
yap, I used it by mistake 👍